src/autopas/containers/verletClusterLists/VerletClusterLists.h

```diff
@@ -703,6 +696,55 @@ class VerletClusterLists : public ParticleContainerInterface<FullParticleCell<Pa
   }
 
  private:
+  /**
+   * Helper function for the region iterators to determine bounds and towers to iterate over.
+   * @param lowerCorner
+   * @param higherCorner
+   * @param behavior
+   * @return
+   */
+  [[nodiscard]] auto getRegionIteratorHelper(const std::array<double, 3> &lowerCorner,
+                                             const std::array<double, 3> &higherCorner,
+                                             IteratorBehavior behavior) const {
+    // Check all cells, as dummy particles are outside the domain they are only found if the search region is outside
+    // the domain.
+    const auto lowerCornerInBounds = utils::ArrayMath::max(lowerCorner, _haloBoxMin);
+    const auto upperCornerInBounds = utils::ArrayMath::min(higherCorner, _haloBoxMax);
+
+    if (not _builder) {
+      // if no builder exists the clusters have not been built yet and all particles are stored in the first tower.
+      return std::make_tuple(lowerCornerInBounds, upperCornerInBounds, std::vector<size_t>{0});
+    }
+
+    // Find towers intersecting the search region
+    auto firstTowerCoords = _builder->getTowerCoordinates(lowerCornerInBounds);
+    auto firstTowerIndex = _builder->towerIndex2DTo1D(firstTowerCoords[0], firstTowerCoords[1]);
+    auto lastTowerCoords = _builder->getTowerCoordinates(upperCornerInBounds);
+
+    std::array<size_t, 2> towersOfInterstPerDim;
+    for (size_t dim = 0; dim < towersOfInterstPerDim.size(); ++dim) {
+      // use ternary operators instead of abs because these are unsigned values
+      towersOfInterstPerDim[dim] = firstTowerCoords[dim] > lastTowerCoords[dim]
+                                       ? firstTowerCoords[dim] - lastTowerCoords[dim]
+                                       : lastTowerCoords[dim] - firstTowerCoords[dim];
```

Comment:
I'm not certain that this is the correct way to do it.

Shouldn't you rather swap `lowerCorner` and `upperCorner` to handle this case?

Currently, it looks like you would at least need to swap `firstTowerCoords` and `lastTowerCoords` as well.

Reply:
Not sure I get what you mean. This is basically 
`towersOfInterstPerDim[dim] = abs(firstTowerCoords[dim] - lastTowerCoords[dim])` 
because we need to know how many towers per dimension are in the area of interest. So we need the difference on the indices in the tower grid and not domain coordinates.

Reply:
lasttowercoords - firsttowercoords should always be positive.
if it is negative, then the for loop over the `iota` should have a negative direction, shouldn't it?

Reply:
I don't think we enforce that the lower so firstTowerCoors is actually not guaranteed to be the first if the user messes up the input.
Since we do not make any guarantees about the order of iteration of the region iterator I would say the direction of the loop does not matter. However, I suppose instead of starting at `firstTowerIndex` we should start at `std::min(firstTowerIndex, lastTowerIndex)`.